src/xenia/kernel/objects/xuser_module.cc

```diff
@@ -104,12 +104,22 @@ X_STATUS XUserModule::LoadFromMemory(const void* addr, const size_t length) {
   // Cache some commonly used headers...
   this->xex_module()->GetOptHeader(XEX_HEADER_ENTRY_POINT, &entry_point_);
   this->xex_module()->GetOptHeader(XEX_HEADER_DEFAULT_STACK_SIZE, &stack_size_);
+  dll_module_ = !!(header->module_flags & XEX_MODULE_DLL_MODULE);
 
   OnLoad();
 
   return X_STATUS_SUCCESS;
 }
 
+X_STATUS XUserModule::Unload() {
+  if (!xex_module()->Unload()) {
```

Comment:
This function should probably have a check to make sure it's not redundant (for manual Unload() followed by the dtor Unload()) -- otherwise anyone listening for OnUnload() will get called twice, possibly.